### PR TITLE
Migrate zipmap unit test to new framework

### PR DESCRIPTION
**Pull request by Shivshankar-Reddy**
Migrate zipmap unit test to new unit test framework, parent ticket #428 .

### CODE REVIEW

File: src/unit/test_zipmap.c [hunk 1]
```diff
@@ -0,0 +1,146 @@
+#include "../zipmap.c"
+#include "test_help.h"
+
+static void zipmapRepr(unsigned char *p) {
+    unsigned int l;
+
+    printf("{status %u}",*p++);
```
Comment (madolson) [inline src/unit/test_zipmap.c]: It looks like you removed all the prints except this one?

Comment (Shivshankar-Reddy) [reply src/unit/test_zipmap.c]: Oh sorry, I missed to remove!

File: src/unit/test_zipmap.c [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    zm = zipmapSet(zm,(unsigned char*) "noval",5, (unsigned char*) "",0,NULL);
+    zipmapRepr(zm);
+    zm = zipmapDel(zm,(unsigned char*) "new",3,NULL);
+    zipmapRepr(zm);
+    unsigned char *value;
+    unsigned int vlen;
+
+    if (zipmapGet(zm,(unsigned char*) "foo",3,&value,&vlen)) {
+        TEST_ASSERT(5 == vlen);
+        TEST_ASSERT(strcmp("12345", (const char*)value));
```
Comment (karthyuom) [inline src/unit/test_zipmap.c]: Overall, it looks good to me, except this assert needs correction as below:
`TEST_ASSERT(!strcmp("12345", (const char*)value));`

**Review by karthyuom — CHANGES_REQUESTED** [review]

File: src/unit/test_zipmap.c (hunk 2 above)
Comment (Shivshankar-Reddy) [reply src/unit/test_zipmap.c]: Thanks for review, I changed it to strncmp instead strcmp.

**Review by karthyuom — APPROVED** [review]
LGTM.

File: src/unit/test_zipmap.c [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+}
+
+int test_zipmapLookUpLargeKey(int argc, char *argv[], int flags) {
+    unsigned char *zm;
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+    zm = zipmapNew();
+
+    zm = zipmapSet(zm,(unsigned char*) "name",4, (unsigned char*) "foo",3,NULL);
```
Comment (hpatro) [inline src/unit/test_zipmap.c]: Could we apply the clang-format on the unit test file as well?

File: src/unit/test_zipmap.c [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    zm = zipmapSet(zm,(unsigned char*) "surname",7, (unsigned char*) "foo",3,NULL);
+    zm = zipmapSet(zm,(unsigned char*) "age",3, (unsigned char*) "foo",3,NULL);
+    zipmapRepr(zm);
```
Comment (hpatro) [inline src/unit/test_zipmap.c:41]: Is this useful/required?

**Comment by hwware** [comment]
@enjoy-binbin Pls help do this pr review, I just rebased it and updated to the latest commit, Thanks

**Review by enjoy-binbin — APPROVED** [review]
This seems ok to me.

**Comment by zuiderkwast** [comment]
We have a failure on the fortify-source build. https://github.com/valkey-io/valkey/actions/runs/10624476718/job/29452862969

Is it the same problem we have seen in #949, which is for GCC 13 only?

**Review by lilpoozie2005 — APPROVED** [review]

**Pull request merged**